File: inventarioBack/src/main/java/com/tienda/inventario/service/ProductoService.java
package com.tienda.inventario.service;

import com.tienda.inventario.model.Producto;
import com.tienda.inventario.repository.ProductoRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor // Crea automáticamente el constructor con las dependencias
public class ProductoService {

    private final ProductoRepository productoRepository;

    // Crear producto
    public Producto crearProducto(Producto producto) {
        return productoRepository.save(producto);
    }

    // Obtener todos los productos
    public List<Producto> obtenerTodos() {
        return productoRepository.findAll();
    }

    // Obtener producto por ID
    public Optional<Producto> obtenerPorId(String id) {
        return productoRepository.findById(id);
    }

    // Actualizar producto
    public Producto actualizarProducto(Producto producto) {
        return productoRepository.save(producto);
    }

    // Eliminar producto
    public void eliminarProducto(String id) {
        productoRepository.deleteById(id);
    }
}

File: tienda-inventario-front/src/services/productoService.js
import axios from 'axios';
const API_URL = 'http://localhost:8080/productos';

// Obtener todos los productos
export const obtenerProductos = async () => {
  try {
    const response = await axios.get(API_URL);
    return response.data;
  } catch (error) {
    console.error('Error al obtener productos:', error);
    return [];
  }
};

// Eliminar un producto por ID
export const eliminarProducto = async (id) => {
  try {
    await axios.delete(`${API_URL}/${id}`);
  } catch (error) {
    console.error('Error al eliminar el producto:', error);
  }
};


export const actualizarProducto = async (id, producto) => {
  try {
    const response = await axios.put(`${API_URL}/${id}`, producto, {
      headers: { 'Content-Type': 'application/json' },
    });
    return response.data;
  } catch (error) {
    console.error('Error al actualizar el producto:', error);
    throw error;
  }
};





